File: src/main/java/za/co/ioagentsmith/game/kalaha/model/Board.java
package za.co.ioagentsmith.game.kalaha.model;

import za.co.ioagentsmith.game.kalaha.util.BoardEnum;

import java.util.stream.IntStream;

public class Board {

    private Pit[] pits;

    public Board() {
        this.pits = new Pit[BoardEnum.TOTAL_PITS.getValue()];

        IntStream.range(0, BoardEnum.TOTAL_PITS.getValue()).forEach(i -> {
            Pit pit = new Pit();
            pits[i] = pit;
            pits[i].setId(i);
            if (((i + 1) % BoardEnum.PITS_PER_PLAYER.getValue()) == 0) {
                pits[i].setKalaha(true);
            } else {
                pits[i].setKalaha(false);
            }
            if (!pits[i].isKalaha()) {
                pits[i].setNumberOfSeeds(BoardEnum.SEEDS_PER_PIT.getValue());
            } else {
                pits[i].setNumberOfSeeds(0);
            }
        });
    }

    public Pit[] getPits() {
        return pits;
    }

    public void setPits(final Pit[] pits) {
        this.pits = pits;
    }

    public Pit getPitById(int pitId) {
        return this.getPits()[pitId];
    }
}

File: kalaha-kotlin-backend/src/main/java/za/co/ioagentsmith/game/kalaha/model/Board.kt
package za.co.ioagentsmith.game.kalaha.model

import za.co.ioagentsmith.game.kalaha.util.BoardEnum

import java.util.stream.IntStream

class Board {

    var pits: MutableList<Pit>? = null

    init {
        pits = mutableListOf()

        IntStream.range(0, BoardEnum.TOTAL_PITS.value).forEach { i ->
            val pit = Pit()
            pits?.add(pit)
            pits!![i].id = i
            if ((i + 1) % BoardEnum.PITS_PER_PLAYER.value == 0) {
                pits!![i].isKalaha = true
            } else {
                pits!![i].isKalaha = false
            }
            if (!pits!![i].isKalaha) {
                pits!![i].numberOfSeeds = BoardEnum.SEEDS_PER_PIT.value
            } else {
                pits!![i].numberOfSeeds = 0
            }
        }
    }

    fun getPitById(pitId: Int): Pit {
        return this.pits!![pitId]
    }
}
